react/features/video-menu/components/web/FakeParticipantContextMenu.tsx

```diff
@@ -0,0 +1,154 @@
+// @ts-ignore
+import { showOverflowDrawer } from '../../../toolbox/functions.web';
+import { hideWhiteboard } from '../../../whiteboard/actions';
+
+type Props = {
+
+    /**
+     * Class name for the context menu.
+     */
+    className?: string;
+
+    /**
+     * Closes a drawer if open.
+     */
+    closeDrawer?: () => void;
+
+    /**
+     * The participant for which the drawer is open.
+     * It contains the displayName & participantID.
+     */
+    drawerParticipant?: {
+        displayName: string;
+        participantID: string;
+    };
+
+    /**
+     * Shared video local participant owner.
+     */
+    localVideoOwner?: boolean;
+
+    /**
+     * Target elements against which positioning calculations are made.
+     */
+    offsetTarget?: HTMLElement;
+
+    /**
+     * Callback for the mouse entering the component.
+     */
+    onEnter?: (e?: React.MouseEvent) => void;
+
+    /**
+     * Callback for the mouse leaving the component.
+     */
+    onLeave?: (e?: React.MouseEvent) => void;
+
+    /**
+     * Callback for making a selection in the menu.
+     */
+    onSelect: (value?: boolean | React.MouseEvent) => void;
+
+    /**
+     * Participant reference.
+     */
+    participant: Participant;
+
+    /**
+     * Whether or not the menu is displayed in the thumbnail remote video menu.
+     */
+    thumbnailMenu?: boolean;
+};
+
+const FakeParticipantContextMenu = ({
+    className,
+    closeDrawer,
+    drawerParticipant,
+    localVideoOwner,
+    offsetTarget,
+    onEnter,
+    onLeave,
+    onSelect,
+    participant,
+    thumbnailMenu
+}: Props) => {
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const _overflowDrawer: boolean = useSelector(showOverflowDrawer);
+
+    const clickHandler = useCallback(() => onSelect(true), [ onSelect ]);
+
+    const _onStopSharedVideo = useCallback(() => {
+        clickHandler();
+        dispatch(stopSharedVideo());
+    }, [ stopSharedVideo ]);
+
+    const _onHideWhiteboard = useCallback(() => {
+        clickHandler();
+        dispatch(hideWhiteboard());
+    }, []);
+
+    const _getFakeParticipantActions = useCallback(() => {
+
+        if (participant.isWhiteboard) {
+            return [ {
+                accessibilityLabel: t('toolbar.hideWhiteboard'),
+                icon: IconShareVideo,
+                onClick: _onHideWhiteboard,
+                text: t('toolbar.hideWhiteboard')
+            } ];
+        }
+
+        if (localVideoOwner) {
+            return [ {
+                accessibilityLabel: t('toolbar.stopSharedVideo'),
+                icon: IconShareVideo,
+                onClick: _onStopSharedVideo,
+                text: t('toolbar.stopSharedVideo')
+            } ];
+        }
+
+        return;
+    }, []);
+
+    return (
+        <ContextMenu
+            className = { className }
+            entity = { participant }
+            hidden = { thumbnailMenu ? false : undefined }
```

Comment:
isn't the outcome the same, since both values are falsey?

Reply:
Oh I just copy pasted this part, I'll take care of it

Reply:
Seems like there is some logic attached to it being undefined or not in the ContextMenu component.